frontend/src/components/workload/Overview.tsx

```diff
@@ -70,7 +70,7 @@ export default function Overview() {
       joint = joint.concat(items);
     }
     return joint;
-  }
+  }, [workloadsData]);
```

Comment:
I don't think this is right... Dependencies only trigger an update if the reference to workloadsData changes. It does not do a deep comparison. From what I see the key/values are updated, and workloadsData keeps its reference. So the reference won't change - meaning this is a stale data bug. Suggest removing the useMemo change.

Reply:
> I don't think this is right... Dependencies only trigger an update if the reference to workloadsData changes. It does not do a deep comparison. From what I see the key/values are updated, and workloadsData keeps its reference. So the reference won't change - meaning this is a stale data bug. Suggest removing the useMemo change.

Hmm, workloadsData is a state. We set it with the useState's setter. We never update only its key/values, from what I can see. What am I missing?

Reply:
Ok, I was mistaken. 

I see workloads is updated with a new reference in here:
```
  function setWorkloads(newWorkloads: WorkloadDict) {
    setWorkloadsData(workloads => ({
      ...workloads,
      ...newWorkloads,
    }));
  }
```